subsys/bluetooth/audio/shell/pbp.c

```diff
@@ -0,0 +1,91 @@
+/** @file
+ *  @brief Bluetooth Public Broadcast Profile shell
+ */
+
+/*
+ * Copyright 2023 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/shell/shell.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/bluetooth/audio/pbp.h>
+
+#include "shell/bt.h"
+
+#define PBS_DEMO                'P', 'B', 'P'
+
+const uint8_t pba_metadata[] = {
+	BT_AUDIO_CODEC_DATA(BT_AUDIO_METADATA_TYPE_PROGRAM_INFO, PBS_DEMO)
+};
+
+enum bt_pbp_announcement_feature pbp_features;
+extern const struct shell *ctx_shell;
+
+static int cmd_pbp_set_features(const struct shell *sh, size_t argc, char **argv)
+{
+	int err = 0;
+	enum bt_pbp_announcement_feature features;
+
+	features = shell_strtoul(argv[1], 16, &err);
+	if (err != 0) {
+		shell_error(sh, "Could not parse received features: %d", err);
+
+		return -ENOEXEC;
+	}
+
+	pbp_features = features;
+
+	return err;
+}
+
+size_t pbp_ad_data_add(struct bt_data data[], size_t data_size)
+{
+	int err;
+
+	/**
+	 * Public Broadcast Announcement Service UUID +
+	 * + Public Broadcast Announcement features + Metadata Size
+	 */
+	NET_BUF_SIMPLE_DEFINE(pbp_ad_buf, BT_UUID_SIZE_16 + 1 + ARRAY_SIZE(pba_metadata));
```

Comment:
This buffer needs to be `static` otherwise the `data[0].data` will point to deallocated memory